client/src/components/RelatedObjectNoteModal.js

```diff
@@ -66,30 +61,63 @@ export default class RelatedObjectNoteModal extends Component {
             submitForm
           }) => {
             const isJson = note.type !== NOTE_TYPE.FREE_TEXT
-            const jsonFields = isJson ? JSON.parse(note.text) : {}
+            const jsonFields = isJson && note.text ? JSON.parse(note.text) : {}
             const noteText = isJson ? jsonFields.text : note.text
+            const typeName =
+              note.type === NOTE_TYPE.PARTNER_ASSESSMENT ? "assessment" : "note"
             return (
               <Form>
                 <Modal.Header closeButton>
                   <Modal.Title>
-                    {note.uuid ? "Edit note" : "Post a new note"}
+                    {(note.uuid ? "Edit " : "Post a new ") + typeName}
                   </Modal.Title>
                 </Modal.Header>
                 <Modal.Body>
-                  <Messages error={this.state.error} />
-                  <Field
-                    name="text"
-                    value={noteText}
-                    component={FieldHelper.renderSpecialField}
-                    onChange={value => setFieldValue("text", value)}
-                    widget={
-                      <RichTextEditor
-                        className="textField"
-                        onHandleBlur={() => setFieldTouched("text", true)}
-                      />
-                    }
-                    vertical
-                  />
+                  <div
+                    style={{
+                      display: "flex",
+                      flexDirection: "column",
+                      padding: 5,
+                      height: "100%"
+                    }}
+                  >
+                    <Messages error={this.state.error} />
+
+                    {note.type === NOTE_TYPE.PARTNER_ASSESSMENT && (
+                      <>
+                        {this.props.questions.map(question => (
+                          <React.Fragment key={question.id}>
+                            <p>{question.label}</p>
+                            <Field
+                              name={question.id}
+                              label=""
+                              component={FieldHelper.renderButtonToggleGroup}
+                              buttons={question.choice}
+                              onChange={value => {
+                                setFieldValue(question.id, value)
+                              }}
+                            />
+                            <br />
+                            <br />
+                          </React.Fragment>
+                        ))}
+                      </>
+                    )}
+
+                    <Field
+                      name="note"
```

Comment:
Why change the name of the field?

Reply:
No good reason, I missed that in the review. Reverted to the original value